packages/aws-cdk-lib/aws-logs/lib/data-protection-policy.ts

```diff
@@ -0,0 +1,375 @@
+}
+
+type FirehoseDestination = {
+  deliveryStream: string;
+}
+
+type S3Destination = {
+  bucket: string;
+}
+
+/**
+ * Interface for creating a data protection policy
+ */
+export interface DataProtectionPolicyProps {
+  /**
+   * Name of the data protection policy
+   *
+   * @default - 'data-protection-policy-cdk'
+   */
+  readonly name?: string;
+
+  /**
+   * Description of the data protection policy
+   *
+   * @default - 'cdk generated data protection policy'
+   */
+  readonly description?: string;
+
+  /**
+   * List of data protection identifiers. Must be in the following list: https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/protect-sensitive-log-data-types.html
+   *
+   * @default - no identifiers
+   */
+  readonly identifiers?: DataIdentifier[];
+
+  /**
+   * For futureproofing; if an identifier is not part of DataIdentifier, but still supported by data protection, a string (full ARN) can be supplied instead.
+   *
+   * @default - no string identifiers
+   */
+  readonly identifierArnStrings?: string[];
+
+  /**
+   * CloudWatch Logs log group name to send audit findings to. The log group must already exist prior to creating the data protection policy.
+   *
+   * @default - no CloudWatch Logs audit destination
+   */
+  readonly logGroupNameAuditDestination?: string;
+
+  /**
+   * S3 bucket name to send audit findings to. The bucket must already exist.
+   *
+   * @default - no S3 bucket audit destination
+   */
+  readonly s3BucketNameAuditDestination?: string;
+
+  /**
+   * Amazon Kinesis Data Firehose delivery stream to send audit findings to. The delivery stream must already exist.
+   *
+   * @default - no firehose delivery stream audit destination
+   */
+  readonly deliveryStreamNameAuditDestination?: string;
+}
+
+
+/**
+ * A data protection identifier. If an identifier is supported but not in this enum, it can be passed as a string instead.
+ */
+export enum DataIdentifier {
+  /** Address */
+  ADDRESS = 'Address',
+  /** AwsSecretKey */
+  AWSSECRETKEY = 'AwsSecretKey',
+  /** BankAccountNumber-DE */
+  BANKACCOUNTNUMBER_DE = 'BankAccountNumber-DE',
+  /** BankAccountNumber-ES */
+  BANKACCOUNTNUMBER_ES = 'BankAccountNumber-ES',
+  /** BankAccountNumber-FR */
+  BANKACCOUNTNUMBER_FR = 'BankAccountNumber-FR',
+  /** BankAccountNumber-GB */
+  BANKACCOUNTNUMBER_GB = 'BankAccountNumber-GB',
+  /** BankAccountNumber-IT */
+  BANKACCOUNTNUMBER_IT = 'BankAccountNumber-IT',
+  /** BankAccountNumber-US */
+  BANKACCOUNTNUMBER_US = 'BankAccountNumber-US',
+  /** CepCode-BR */
+  CEPCODE_BR = 'CepCode-BR',
+  /** Cnpj-BR */
+  CNPJ_BR = 'Cnpj-BR',
+  /** CpfCode-BR */
+  CPFCODE_BR = 'CpfCode-BR',
+  /** CreditCardExpiration */
+  CREDITCARDEXPIRATION = 'CreditCardExpiration',
+  /** CreditCardNumber */
+  CREDITCARDNUMBER = 'CreditCardNumber',
+  /** CreditCardSecurityCode */
+  CREDITCARDSECURITYCODE = 'CreditCardSecurityCode',
+  /** DriversLicense-AT */
+  DRIVERSLICENSE_AT = 'DriversLicense-AT',
```

Comment:
Given the different types of countries supported by each identifier type, I think adding this would introduce too many enums and be confusing to customers. https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/protect-sensitive-log-data-types.html

Reply:
Why would more enums make it confusing to customers? The way I am looking at it is when a user is using autocomplete in their editor they would see what type of `DataIdentifier`s are available, they would pick one `driversLicense` and then see individual options for that identifier. The current solution is to see a super long list that could continue to grow longer and longer.

Reply:
The only way I can see this working is adding an enum for each of the 33 data identifiers that have a country code suffix, since the list of supported countries is different for each identifier. Is that what you're suggesting? Otherwise, if we have this only for `driversLicense` it wouldn't be consistent with the other identifiers. When the user types `DRIVERSLICENSE` the autocomplete should also be clear to select one with a country code.

Reply:
Ok you've convinced me 👍